float32[4]  current
float32[4]  position
float32[4]  speed